docs/git/cheat-sheet.md

```diff
@@ -0,0 +1,67 @@
+# Git Cheat Sheet
+
+The following is a cheat sheet for using [git] while developing Cardano SL
+which should contain all the git operations that someone working on Cardano
+will need at some point.
+
+### Cloning the Cardano SL repository
+```
+git clone https://github.com/input-output-hk/cardano-sl
+```
+After you change directory into that cloned repository, you may want to tell
+git to use the SSH protocol for the "git push" command, keeping the HTTPS
+protocol for the "git pull" and "git fetch" commands by issuing the following
+command:
+
+```
+git remote set-url --push origin git@github.com:input-output-hk/cardano-sl
+```
```

Comment:
We can _achieve_ this automatically with a simple trick in the global configuration:

```
[url "git@github.com:"]
  insteadOf = https://github.com/
```

Or simply directly clone the repository using the SSH-URL:

```
git clone git@github.com:input-output-hk/cardano-sl
```

I don't see any reason to favor HTTPs over SSH (even for `fetch` and `pull`), is there anything I am not aware of regarding these :thinking: ?

Reply:
Having a `git@github.com` as the default URL means that people who have not registered an SSH key with Github cannot clone your repository. I do this all the time, if I want to build in a VM and I *don't* want to copy my usual SSH key in the VM?

Reply:
I have a relatively short timeout on my ssh-agent. If I use the `git@github` version of the URL, i have to enter my password for "git pull" operations. With a separate "git push" URL, I only have to enter my password for push operations.

Reply:
`https` will work from any public wifi (e.g. from a caffe shop, etc), where `git-ssh` might be blocked. Said that, I am using `ssh` mostly from home or a local cowork.